### PR TITLE
Modified the JavaDoc to better suit the returning values of the function Request.filterWith

### PR DESCRIPTION
**Pull request by Mathreyu**
As requested by PR [#1318](https://github.com/junit-team/junit4/issues/1318) added and extra to the description to better suit the return values by this function. 

Lest me know if it requires further tweaking :)

### CODE REVIEW

File: src/main/java/org/junit/runner/Request.java [hunk 1]
```diff
@@ -131,8 +131,11 @@ public Request filterWith(Filter filter) {
     }
 
     /**
-     * Returns a Request that only runs contains tests whose {@link Description}
-     * equals <code>desiredDescription</code>
+     * Returns a Request that only runs tests whose {@link Description}
+     * matches <code>desiredDescription</code>
+     *
+     * Returns an empty Request if <code>desiredDescription</code> is not a single test and filters all but the single
```
Comment (kcooney) [inline src/main/java/org/junit/runner/Request.java]: Use `<p>` and `@code`:

    * <p>Returns an empty {@code Request} if {@code desiredDescription} is not...

Comment (Mathreyu) [inline src/main/java/org/junit/runner/Request.java]: sure thing, sending changes.

Comment (Mathreyu) [inline src/main/java/org/junit/runner/Request.java]: The thing is this doesnt seem to work on the Intellij IDEA quick documentation view, where are you checking this changes? or how do you verify the links are displayed?

Comment (kcooney) [inline src/main/java/org/junit/runner/Request.java]: `@code` doesn't generate links. In this case links aren't needed.

Comment (Mathreyu) [inline src/main/java/org/junit/runner/Request.java]: ok, the corrections were done in the commit bellow :)

**Comment by kcooney** [comment]
Thanks! Merged with minor changes.

**Pull request merged**